Documentation/VSCodeConfiguration.md

```diff
@@ -4,20 +4,57 @@ Visual Studio Code does not work optimally for Serenity development, and there's
 
 The WSL Remote extension allows you to use VSCode in Windows while using the normal WSL workflow. This works surprisingly well, but for code comprehension speed you should put the Serenity directory on your WSL root partition.
 
-## Code comprehension
+- [Visual Studio Code Project Configuration](#visual-studio-code-project-configuration)
+  - [Code comprehension](#code-comprehension)
+    - [clangd](#clangd)
+      - [GCC](#gcc)
+      - [Clang](#clang)
+    - [Microsoft C/C++ tools](#microsoft-cc-tools)
+  - [DSL syntax highlighting](#dsl-syntax-highlighting)
+  - [Formatting](#formatting)
+  - [Settings](#settings)
+  - [Customization](#customization)
+    - [Custom Tasks](#custom-tasks)
+    - [License snippet](#license-snippet)
 
-Both C++ comprehension tools listed below report fake errors.
+## Code comprehension
 
 ### clangd
 
-The official clangd extension can be used for C++ comprehension. You'll have to use the following .clangd:
+The clangd extension (`llvm-vs-code-extensions.vscode-clangd`) can be used for C++ comprehension.
+
+You'll have to use the following `.clangd` file for the toolchain you're using:
+
+#### GCC
 
 ```yaml
 CompileFlags:
-  CompilationDatabase: Build/i686
+   Add:
+      [
+         -D__serenity__,
+         -isystem../../Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0,
+         -isystem../../Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0/bits,
+         -isystem../../Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0/i686-pc-serenity,
+      ]
+   CompilationDatabase: Build/i686
+
+```
+
+#### Clang
+
+```yaml
+CompileFlags:
+   Add:
+      [
+         -D__serenity__,
+         -isystem../../Toolchain/Local/clang/include/c++/v1,
+         -isystem../../Toolchain/Local/clang/include/x86_64-pc-serenity/c++/v1,
+      ]
+   CompilationDatabase: Build/x86_64clang
```

Comment:
In the GCC Toolchain you use the i686 config as the example,
Wouldn't it be better to be consistent?

maybe add a sentence about the option of x86_64/aarch64

Reply:
I can't make a more detailed clangd per-arch&toolchain configuration as I didn't test those myself and I rely on information people on the Discord gave me so I can't add detailed information about those arch&toolchain combo

Reply:
I can confirm that this *does* work the same way for all architectures, you just have to substitute `i686`/`x86_64` in both the `CompilationDatabase` path and the include paths for the architecture you want to use. Adding a note about that is a great idea.